pkg/provider/kubernetes/gateway/client.go

```diff
@@ -675,20 +683,20 @@ func translateNotFoundError(err error) (bool, error) {
 	return err == nil, err
 }
 
-func gatewayStatusEquals(statusA, statusB gatev1.GatewayStatus) bool {
+func gatewayStatusEqual(statusA, statusB gatev1.GatewayStatus) bool {
```

Comment:
@kevinpollet shouldn't this also check for addresses? I debugged the problem I was talking to you about [yesterday](https://github.com/traefik/traefik/pull/10667) down to this function. I debugged cloning the repo, building my image, and running Traefik with additional logs.

I ended up always having the gateway.statusAddress to null because this function avoids the update but the two gateway objects are not equal, one has the address set from the service/ip/hostname and the other does not. This function seems not to take into account that so it never updates the gateway with an address, breaking external-dns.

Can you please have a look?

Reply:
Hello @ilmax,

You are right, after checking the code we opened https://github.com/traefik/traefik/pull/10972 to fix it.

Reply:
Thank you for the quick fix @kevinpollet!